### PR TITLE
Adding simple CI

### PR DESCRIPTION
**Pull request by SebaLukas**
## Describe your changes
Added lint, build and test workflows.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: .ci/build-kit/build_and_test.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+set -e
+
+cmake \
+    -B build \
+    -S "$EXT_MOUNT/source" \
+    -G Ninja \
+    -DISO15118_BUILD_TESTS=ON \
```
Comment (andistorm) [inline .ci/build-kit/build_and_test.sh:9]: As already mentioned this flag should be changed somewhen according https://everest.github.io/nightly/general/faq.html#testing

File: .github/workflows/build_and_test.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          docker run \
+          --volume "$(pwd):/ext" \
+          --name test-container \
+          build-kit run-script build_and_test
+      - name: Archive test results
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: ctest-report
+          path: ${{ github.workspace }}/ctest-report
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:56]:
```suggestion
          path: ${{ github.workspace }}/ctest-report

```

**Review by andistorm — APPROVED** [review]
LGTM

File: .ci/build-kit/build_and_test.sh (hunk 1 above)
Comment (SebaLukas) [reply .ci/build-kit/build_and_test.sh:9]: Yes, I will change this when I adjust the CMakeLists.txt

**Pull request merged**